app/controllers/resources_controller.rb

```diff
@@ -28,35 +29,80 @@ def create
            status: :created
   end
 
+  # PUT /resource/:id
+  def update
+    authorize! :resource
+
+    begin
+      request_dro = cocina_model(params.except(:action, :controller, :resource, :id).to_unsafe_h)
+    rescue BlobError => e
+      # Returning 500 because not clear whose fault it is.
+      return render build_error('500', e, 'Error matching uploading files to file parameters.')
+    end
+
+    result = BackgroundJobResult.create(output: {})
+    UpdateJob.perform_later(model_params: JSON.parse(request_dro.to_json), # Needs to be sidekiq friendly serialization
+                            signed_ids: signed_ids(params),
+                            background_job_result: result)
+
+    render json: { jobId: result.id },
+           location: result,
+           status: :created
+  end
+
   private
 
   def cocina_model(model_params)
     new_model_params = model_params.deep_dup
-    new_model_params[:version] = 1
     decorate_file_sets(new_model_params)
+    Cocina::Models::DRO.new(new_model_params)
+  end
+
+  def cocina_request_model(model_params)
+    new_model_params = model_params.deep_dup
+    new_model_params[:version] = 1
+    decorate_request_file_sets(new_model_params)
 
     Cocina::Models::RequestDRO.new(new_model_params)
   end
 
   # Decorates the provided FileSets with the information we have in the ActiveStorage table.
   # externalIdentifier is also removed from the request.
-  # rubocop:disable Metrics/AbcSize
   def decorate_file_sets(model_params)
     file_sets(model_params).each do |fileset|
-      fileset[:version] = 1
+      fileset[:version] = model_params[:version]
```

Comment:
Fileset version changes even if files haven't?

Reply:
Yes, this is synched to the parent, since we don't track file versions separately in Fedora. I think this could change when we move to a different data store.